### ISSUE TITLE
Trash doesn't work in Atom, Nemo, Nautilus

### ISSUE THREAD

**Issue by papojari** [NixOS/nixpkgs#128026]
**Describe the bug**
The trash in several of my gui applications doesn't work anymore since some time. Atom asks if `gvfs-trash` is installed, nemo doesn't show the trash, Favorites and Recents anymore in it's left panel and Nautilus complains that it could not display `trash:///`. If I open nemo through me terminal though (no error messages) everything works. Normally I launch programs in sway with this command: `wofi --show=drun --lines=10 --prompt=""`. Wofi is a gui launcher.

**To Reproduce**
1. Install NixOS from the GNOME ISO
2. Install Sway, Wofi, Atom, Nemo and Nautilus.
3. Try to move files to the trash in Atom, Nemo or Nautilus.

**Expected behavior**
Files get moved to the trash. Nemo shows Trash, Favorites and Recents.

**Screenshots**
- Atom
![swappy-20210624_181121](https://user-images.githubusercontent.com/81317317/123298295-a50dab80-d518-11eb-865d-93a94322dc9c.png)

- Nautilus
![swappy-20210624_181338](https://user-images.githubusercontent.com/81317317/123298313-a8089c00-d518-11eb-8617-a73e6077e969.png)

**Additional context**
I am on the unstable branch.

**Notify maintainers**

- `gvfs` 
  - @hedning 
  - @dasj19
  - @maxeaubrey 
- `wofi`
  - @Elyhaka
- `cinnamon.nemo`
  - @mkg20001 
- `atom`
  - @offlinehacker 
  - @ysndr 
- `gnome.nautilus`
  - *maintainers are already meantioned in the `gvfs` package*

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
these paths will be fetched (0.05 MiB download, 0.28 MiB unpacked):
  /nix/store/lp90qhw19dcp79cnlicjg09llgh7dzsz-bash-interactive-4.4-p23-dev
copying path '/nix/store/lp90qhw19dcp79cnlicjg09llgh7dzsz-bash-interactive-4.4-p23-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.44, NixOS, 21.11pre297244.d2b489b24ae (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixos-21.11pre297244.d2b489b24ae, nixpkgs-21.11pre297600.65db0350fe3"`
 - channels(papojari): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by TredwellGit** [comment]
Have you declared `programs.sway.wrapperFeatures.gtk = true;`?

**Comment by papojari** [comment]
> Have you declared `programs.sway.wrapperFeatures.gtk = true;`?

yes

**Comment by papojari** [comment]
In GNOME 40 everything works btw. So this is something with sway?

**Comment by TredwellGit** [comment]
Try something like `set $menu bash -ic 'wofi --show=drun --lines=10 --prompt="" | xargs swaymsg exec --'`. You are probably experiencing https://lwn.net/Articles/709769/.

**Comment by papojari** [comment]
Thanks that fixes Nemo and Nautilus.

**Comment by papojari** [comment]
This issue only occurs when gdm is used as the display manager. When starting sway in a shell with `sway` everything works! I should have listened to the sway wiki not recommending gdm. 🤦‍♀️

**Comment by berbiche** [comment]
I was already using the two proposed solutions before finding this issue.

This is personally fixed by #149724 + running gvfsd.
